src/job-cache/job-cache.cpp

```diff
@@ -968,6 +1171,22 @@ wcl::optional<MatchingJob> Cache::read(const FindJobRequest &find_request) {
     output_file.path = wcl::join_paths(*pair.first, rel_path);
   }
 
+  for (auto &output_dir : result->output_dirs) {
+    std::vector<std::string> output_path_vec = split_path(output_dir.path);
+    auto pair = find_request.dir_redirects.find_max(output_path_vec.begin(), output_path_vec.end());
+    if (!pair.first) continue;
+    std::string rel_path = join('/', pair.second, output_path_vec.end());
+    output_dir.path = wcl::join_paths(*pair.first, rel_path);
+  }
+
+  for (auto &output_symlink : result->output_symlinks) {
+    std::vector<std::string> output_path_vec = split_path(output_symlink.path);
+    auto pair = find_request.dir_redirects.find_max(output_path_vec.begin(), output_path_vec.end());
+    if (!pair.first) continue;
+    std::string rel_path = join('/', pair.second, output_path_vec.end());
+    output_symlink.path = wcl::join_paths(*pair.first, rel_path);
+  }
+
```

Comment:
what is this code doing?

Reply:
This is one of those times you're baiting me to add a comment huh ;)

These rename sandbox paths into non-sandbox paths. The paths that I expect in the json are sandbox absolute paths and thats also what I store in the database. The paths outside the sandbox are generally relative. You can add redirects to redirect any sandbox path to a non-sandbox location. For instance you can redirect "/workspace" to "./"

Reply:
I added comments, let me know if that helps.

Reply:
eh goal wasn't directly to bait you into writing a comment but I did have trouble following what it was doing so a comment or function extraction felt reasonable